packages/block-library/src/post-title/edit.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityProp, useEntitySaving } from '@wordpress/core-data';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { RichText } from '@wordpress/block-editor';
+import { cleanForSlug } from '@wordpress/editor';
+
+const saveProps = [ 'title', 'slug' ];
+export default function PostTitleEdit() {
+	const [ title, setTitle ] = useEntityProp( 'postType', 'post', 'title' );
+	const [ , setSlug ] = useEntityProp( 'postType', 'post', 'slug' );
```

Comment:
The hard-coding of `post` here is probably just temporary, but we need to figure out a way to do it right, since currently this only works for posts of post type 'post' and causes an error on other post types.

The problem is that the overall current post being edited appears to be overridden by this implementation, causing `getCurrentPost()` to return the `wp_template` instead of the actual post being edited. Otherwise we could have used `withSelect()` to get the current post type, but that wouldn't work.

I'm a bit wary of overriding the current post context in Gutenberg to become a `wp_template` post when actually editing a "real" post, since it could break expectations by other plugins. We should probably think about a way we expose the current `wp_template` through a set of new API functions (e.g. `getCurrentTemplate()` etc.), where we could transform the editor accordingly based on whether these include a `wp_template` post or not (if not, it would be just like today).

Reply:
The editor already supports editing multiple post types and the editor store has a `getPostType` selector.

Reply:
Are you referring to `getCurrentPostType()`? It still returns `wp_template` here when I use it. How do I get the post type of the actual post being edited? It's hard-coded to `post` here, so we need to replace it with a function that dynamically gets the correct post type.

Reply:
Sorry, I thought you were asking how to get the post type of the template. (The template is the actual post being edited after all.)

You can get the post that uses the template through editor settings:

```js
const { postId, postType, templateId } = getEditorSettings()
```